tools/cli/factory.go

```diff
@@ -95,16 +101,65 @@ func (b *clientFactory) SDKClient(c *cli.Context, namespace string) sdkclient.Cl
 	return sdkClient
 }
 
-func (b *clientFactory) createGRPCConnection(hostPort string) *grpc.ClientConn {
+func (b *clientFactory) createGRPCConnection(c *cli.Context) *grpc.ClientConn {
+	hostPort := c.GlobalString(FlagAddress)
 	if hostPort == "" {
 		hostPort = localHostPort
 	}
+	// Ignoring error as we'll fail to dial anyway, and that will produce a meaningful error
+	host, _, _ := net.SplitHostPort(hostPort)
+
+	certPath := c.GlobalString(FlagTLSCertPath)
+	keyPath := c.GlobalString(FlagTLSKeyPath)
+	caPath := c.GlobalString(FlagTLSCaPath)
+
+	grpcSecurityOptions := grpc.WithInsecure()
+	var cert *tls.Certificate
+	var caPool *x509.CertPool
+
+	if caPath != "" {
+		caCertPool, err := fetchCACert(caPath)
+		if err != nil {
+			b.logger.Fatal("Failed to load server CA certificate", zap.Error(err))
+			return nil
```

Comment:
i know the above link is log.Fatal, but should this line be `return err`?

Reply:
This basically the same thing. If this func returns `error` caller will do log anyway. I think it is ok to log from here. Actually I would also add info level logging in case of success. Like `certs are loaded successfully, using TLS mode` or something like this.

Reply:
> This basically the same thing.

yeah, just a little bit allergic about this:

```go
if err != nil {
    return nil
}
```

Reply:
i would expect 
```go
if err != nil {
    return err
}
```

Reply:
>Actually I would also add info level logging in case of success. Like certs are loaded successfully, using TLS mode or something like this.

Since this is a command line tool, I think it'd be too noisy to log a "connected successfully" every time it is executed.